utils/controller_debug.py

```diff
@@ -0,0 +1,789 @@
+        else:
+            data = [int(x, 0) for x in param[1:]]
+        poke(addr, data, ptype)
+
+    def do_console(self, line):
+        """
+Switch from command mode to a simple console display which
+continuously readings debug print statements from the controller
+and displays the data received to the screen.
+
+Enter <ctrl>C to exit this mode
+
+A couple optional parameters can be passed on the command line:
+
+--flush     If given, the console will be flushed before we start
+            to display data.  That way any old (possibly incomplete
+            data) will be removed before the display starts
+
+<filename>  If a file name is given, the data received will be
+            written to the file as well as displayed.
+"""
+
+        cl = line.split()
+
+        # Optionally flush the buffer before we start displaying
+        if "--flush" in cl:
+            while len(SendCmd(OP_PBREAD)) > 0:
+                pass
+            cl.remove("--flush")
+
+        if len(cl) > 0:
+            fp = open(cl[0], "w")
+        else:
+            fp = None
+
+        try:
+            while True:
+                dat = SendCmd(OP_PBREAD)
+                if len(dat) < 1:
+                    time.sleep(0.05)
+                    continue
+
+                S = "".join([chr(x) for x in dat])
+
+                sys.stdout.write(S)
+                sys.stdout.flush()
+
+                if fp != None:
+                    fp.write(S)
+
+        except KeyboardInterrupt:
+            print()
+            ReSync()
+            pass
+
+    def do_EOF(self, line):
+        return True
+
+    def do_exit(self, line):
+        return True
+
+    def do_get(self, line):
+        cl = line.split()
+        if len(cl) < 1:
+            print("Please give the variable name to read")
+            return
+
+        print(GetVar(cl[0]))
+
+    def help_get(self):
+        global varDict
+        print("Read the value of a debug variable and display it\n")
+        print("Variables currently defined:")
+        for k in varDict.keys():
+            print("   %-10s - %s" % (k, varDict[k].help))
+
+    def do_set(self, line):
+        cl = line.split()
+        if len(cl) < 2:
+            print("Please give the variable name and value")
+            return
+        SetVar(cl[0], cl[1])
+
+    def help_set(self):
+        global varDict
+        print("Set the value of a debug variable\n")
+        print("Variables currently defined:")
+        for k in varDict.keys():
+            print("   %-10s - %s" % (k, varDict[k].help))
+
+    # Read info about all the supported variables and load
+    # them in a map
+    def GetVarInfo(self):
+        global varDict
+        varDict = {}
+
+        try:
+            for vid in range(256):
+                dat = SendCmd(OP_VAR, [0] + Split16(vid))
+                V = VarInfo(vid, dat)
+                varDict[V.name] = V
+        except Error as e:
+            pass
+
+
+varDict = {}
```

Comment:
In the function GetVarInfo I'm removing any old contents of that dictionary and repopulating it.
Right now that only gets called once at startup, but it may in the future get called if we detect an update to the firmware which might change the variables that are available.